common/djangoapps/course_modes/models.py

```diff
@@ -86,6 +86,11 @@ class Meta:
         """ meta attributes of this model """
         unique_together = ('course_id', 'mode_slug', 'currency')
 
+    def save(self, force_insert=False, force_update=False, using=None):
```

Comment:
`clean` is only used automatically when working with ModelForms. Overriding `save` is the only implementation I am aware of that guarantees `currency` is lowercase, regardless of how the model is saved.